### PR TITLE
Adds discovery points to mutated plants

### PR DESCRIPTION
**Pull request by desluOs**
## About The Pull Request
This adds the ability to scan mutated plants that have been harvested for 300 discovery points.

## Why It's Good For The Game
Discovery research, especially during lowpop hours, is very reliant on having a living discovery team. This is not something that is always available. This adds a new method for points to be obtained on station via discovering new mutated plants.

## Changelog
:cl:
add: Scanning a harvested mutated plant adds discovery points
/:cl:

### CODE REVIEW

**Review by PowerfulBacon — COMMENTED** [review]
If you are having the discovery_points var on all plants, then the discoverable component should be added in the initialize of the plant superclass if the discovery point value isn't 0.

**Comment by PowerfulBacon** [comment]
Also I like this since scientists gotta talk with the botanists for plants 👍

**Comment by desluOs** [comment]
> If you are having the discovery_points var on all plants, then the discoverable component should be added in the initialize of the plant superclass if the discovery point value isn't 0.

I actually had this originally.  Do you mean putting it in the initialize of grown.dm? Because I didn't want plants that aren't mutated to start the scanning sequence at all if clicked on with a discovery scanner, just specific plants.

If not, which plant superclass are you referring to.

**Comment by PowerfulBacon** [comment]
> > If you are having the discovery_points var on all plants, then the discoverable component should be added in the initialize of the plant superclass if the discovery point value isn't 0.
> 
> I actually had this originally. Do you mean putting it in the initialize of grown.dm? Because I didn't want plants that aren't mutated to start the scanning sequence at all if clicked on with a discovery scanner, just specific plants.
> 
> If not, which plant superclass are you referring to.

Yea, if you want only mutated plants to have the component, set the discovery_point value to 0 on default, and to 300 or whatever on mutated plants and only add the component if discovery_points isn't 0.

**Comment by desluOs** [comment]
Ok I changed it to how PowerfulBacon suggested, and now it should work exactly the same but without filling up all the plant files with initialize() setups.

File: code/modules/hydroponics/grown.dm [hunk 1]
```diff
@@ -45,6 +47,9 @@
 		seed.prepare_result(src)
 		transform *= TRANSFORM_USING_VARIABLE(seed.potency, 100) + 0.5 //Makes the resulting produce's sprite larger or smaller based on potency!
 		add_juice()
+	
+	if(discovery_points && discovery_points > 0)
```
Comment (Kapu1178) [inline code/modules/hydroponics/grown.dm]:
```suggestion
	if(discovery_points)
```

File: code/modules/hydroponics/growninedible.dm [hunk 2]
```diff
@@ -31,6 +32,8 @@
 		transform *= TRANSFORM_USING_VARIABLE(seed.potency, 100) + 0.5
 		add_juice()
 
+	if(discovery_points && discovery_points > 0)
```
Comment (Kapu1178) [inline code/modules/hydroponics/growninedible.dm]:
```suggestion
	if(discovery_points)
```

**Review by Kapu1178 — CHANGES_REQUESTED** [review]

File: code/modules/hydroponics/grown.dm (hunk 1 above)
Comment (Kapu1178) [reply code/modules/hydroponics/grown.dm]: if([thing]) 
if [thing] isnt null (ie: set to a positive integer), it will always be true.

Comment (desluOs) [reply code/modules/hydroponics/grown.dm]: Ah that makes sense, I will make the change to both instances when I get home

**Comment by desluOs** [comment]
Would love to hear more feedback for this : )

**Comment by PerishedFraud** [comment]
how about cargo bounties that give discovery points instead of money

**Comment by desluOs** [comment]
> how about cargo bounties that give discovery points instead of money

Sure, but im just trying to get this one thing added first.  Let me know how you feel about the amount of points or anything like that.

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**